src/widgets/notes/visit-notes-form.component.tsx

```diff
@@ -0,0 +1,371 @@
+  const [
+    currentSessionProviderUuid,
+    setCurrentSessionProviderUuid
+  ] = React.useState<string | null>("");
+  const [
+    currentSessionLocationUuid,
+    setCurrentSessionLocationUuid
+  ] = React.useState("");
+  const [locationUuid, setLocationUuid] = React.useState<string | null>(null);
+  const [providerUuid, setProviderUuid] = React.useState<string | null>(null);
+  const [searchTerm, setSearchTerm] = React.useState("");
+  const [searchResults, setSearchResults] = React.useState<Array<Diagnosis>>(
+    null
+  );
+  const [selectedDiagnoses, setSelectedDiagnoses] = React.useState<
+    Array<Diagnosis>
+  >([]);
+  const [visitDateTime, setVisitDateTime] = React.useState(new Date());
+  const searchInputRef = React.useRef(null);
+
+  closeWorkspace = closeWorkspace ?? (() => switchTo("workspace", ""));
+
+  React.useEffect(() => {
+    if (searchTerm) {
+      const sub = fetchDiagnosisByName(searchTerm).subscribe(
+        results => setSearchResults(results),
+        error => {
+          setError(error);
+          createErrorHandler();
+        }
+      );
+      return () => sub.unsubscribe();
+    } else {
+      setSearchResults(null);
+    }
+  }, [searchTerm]);
+
+  React.useEffect(() => {
+    const ac = new AbortController();
+    fetchCurrentSessionData(ac).then(({ data }) => {
+      const { currentProvider, sessionLocation } = data;
+      if (currentProvider.uuid) {
+        setCurrentSessionProviderUuid(currentProvider.uuid);
+      }
+      if (sessionLocation.uuid) {
+        setCurrentSessionLocationUuid(sessionLocation.uuid);
+      }
+    });
+    return () => ac.abort();
+  }, []);
+
+  React.useEffect(() => {
+    const ac = new AbortController();
+    if (currentSessionProviderUuid) {
+      fetchProviderByUuid(ac, currentSessionProviderUuid).then(({ data }) => {
+        setProviderUuid(data.uuid);
+      });
+    }
+  }, [currentSessionProviderUuid]);
+
+  React.useEffect(() => {
+    const ac = new AbortController();
+    if (currentSessionLocationUuid) {
+      fetchLocationByUuid(ac, currentSessionLocationUuid).then(({ data }) =>
+        setLocationUuid(data.uuid)
+      );
+    }
+  }, [currentSessionLocationUuid]);
+
+  const handleCancel = () => {
+    closeWorkspace();
+  };
+
+  const handleSearchTermChange = debounce(searchTerm => {
+    setSearchTerm(searchTerm);
+  }, searchTimeoutInMs);
+
+  const resetSearch = () => {
+    setSearchTerm("");
+    setSearchResults([]);
+  };
+
+  const handleAddDiagnosis = (diagnosisToAdd: Diagnosis) => {
+    resetSearch();
+    setSelectedDiagnoses(selectedDiagnoses => [
+      ...selectedDiagnoses,
+      diagnosisToAdd
+    ]);
+  };
+
+  const handleRemoveDiagnosis = (diagnosisToRemove: Diagnosis) => {
+    setSelectedDiagnoses(
+      selectedDiagnoses.filter(
+        diagnosis => diagnosis.concept.id !== diagnosisToRemove.concept.id
+      )
+    );
+  };
+
+  const RenderSearchResults: React.FC = () => {
```

Comment:
Another component defined within a component? Either leave this stuff inline in the `render` block or factor it all the way out of the component (with `searchResults`, `handleAddDiagnosis`, and whatever else as props).

Reply:
Sorry, refactored this while tired. Fixed.

Reply:
No worries, thanks!